### PR TITLE
Remaining

### PR DESCRIPTION
**Pull request by seanmtracey**
Cosmetic fixes to labs-sharing

### CODE REVIEW

File: src/js/share.js [hunk 1]
```diff
@@ -163,6 +172,15 @@
 				});
 			}
 
+			getRemainingNumberOfTokensForUser()
+				.then(function(amount){
+					rootEl.querySelectorAll('[data-labs-o-share-credit-count]')[0].textContent = amount;
+				})
+				.catch(function(err){
+					console.error("There was an error trying to obtain the remaining number of sharing tokens for the subscriber", err);
```
Comment (JakeChampion) [inline src/js/share.js]: I don't think we should log the error in the console

File: src/js/share.js [hunk 2]
```diff
@@ -352,33 +369,66 @@
 
 		return shareUrlPromises[maxShares];
 	}
+	
+	function getUserUUID(){
```
Comment (JakeChampion) [inline src/js/share.js]: No need for this function, share-code-web will handle this via the user's session

File: src/js/share.js [hunk 3]
```diff
@@ -352,33 +369,66 @@
...
+				.then(function(res){
+					return res.headers.get('FT-User-UUID');
+				})
+	}
 
 	function getRemainingNumberOfTokensForUser(){
-		fetch(serviceURL + "/remainingamount")
-			.then(function(res){
+		// Let's get the user ID
+		return getUserUUID()
```
Comment (JakeChampion) [inline src/js/share.js]:
```
return fetch(serviceURL + "/remainingamount",
  {
    credentials: 'include'
  })
  .then(res => res.json())
  .then(data => data.remainingCredits);
```

File: src/js/share.js [hunk 4]
```diff
@@ -352,33 +369,66 @@
...
 
 	Share.addShareCodeToUrl = function () {
-		if (urlAlreadyHasShareCode(window.location.href) === false) {
-			if (tokenTimeout === undefined) {
-				tokenTimeout = setTimeout(function () {
-					getShareUrl(1)
-					.then(function (data) {
-						if (data.success) {
-							const code = data.data.shareCode;
+		if (urlAlreadyHasShareCode(window.location.href)) {
```
Comment (JakeChampion) [inline src/js/share.js]: Could pass in `window.location.search`

File: src/js/share.js [hunk 5]
```diff
@@ -352,33 +369,66 @@
...
+			let otherParameters = removeExisingShareCodeFromURL();
+			let newURL = window.location.href.split('?')[0];
 
-							const join = (window.location.href.indexOf("?") > -1) ? "&" : "?";
+			if(otherParameters !== ""){
+				newURL += "?" + otherParameters;
```
Comment (JakeChampion) [inline src/js/share.js]: `queryString.stringify` might be useful here

File: src/js/share.js [hunk 6]
```diff
@@ -352,33 +369,66 @@
...
-						}
-					});
-				}, 5000);
+			window.history.pushState({}, undefined, newURL);
 
-			}
+		}
+
+		if (tokenTimeout === undefined) {
+			tokenTimeout = setTimeout(function () {
```
Comment (JakeChampion) [inline src/js/share.js:394]: Why is this in a setTimeout? I can't remember the reasoning

File: src/js/share.js (hunk 3 above)
Comment (seanmtracey) [inline src/js/share.js]: Would we not want to check the success status of the request?

Comment (JakeChampion) [inline src/js/share.js]: Sure, this also raises a good point about the share store.

File: src/scss/share.scss [hunk 7]
```diff
@@ -164,3 +164,25 @@
...
+
+}
+
+.o-overlay-trigger[data-o-overlay-gift-trigger]{
+	cursor:pointer;
+	display: inline-block;
+	margin-top: 2px;
+}
+
+.article__share{
```
Comment (JakeChampion) [inline src/scss/share.scss]: This seems specific to the Next product

**Pull request merged**